tests/cpp_api_tests/rocAL_audio_unittests/rocAL_audio_unittests.cpp

```diff
@@ -200,13 +225,25 @@ int test(int test_case, const char *path, int qa_mode, int downmix, int gpu) {
         for (uint idx = 0; idx < output_tensor_list->size(); idx++) {
             buffer = static_cast<float *>(output_tensor_list->at(idx)->buffer());
             output_tensor_list->at(idx)->copy_roi(roi.data());
-            frames = roi[idx * 4 + 2];
+            if (output_tensor_list->at(idx)->dims().at(2) == 1)
+            {
+                max_samples = 1;
+                max_channels = 1;
+                frames = roi[idx * 4 + 2];
+                buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
+            }
+            else {
+                max_samples = output_tensor_list->at(idx)->dims().at(1);
+                max_channels = output_tensor_list->at(idx)->dims().at(2);
+                frames = roi[idx * 4 + 2];
+                buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
+            }
```

Comment:
```suggestion
            if (output_tensor_list->at(idx)->dims().at(2) == 1)
            {
                max_samples = 1;
                max_channels = 1;
                frames = roi[idx * 4 + 2];
                buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
            }
            else {
                max_samples = output_tensor_list->at(idx)->dims().at(1);
                max_channels = output_tensor_list->at(idx)->dims().at(2);
                frames = roi[idx * 4 + 2];
                buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
            }
```


@HazarathKumarM I would suggest the following here 

```
max_samples = output_tensor_list->at(idx)->dims().at(1);
max_channels = max_channels = output_tensor_list->at(idx)->dims().at(2)
frames = roi[idx * 4 + 2];
buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
```

Instead of all the if else conditions, Please let me know if this will work

Reply:
Also add comment stating why you are setting to 1